targets/psa_switch/psa_switch.cpp

```diff
@@ -396,6 +396,20 @@ PsaSwitch::egress_thread(size_t worker_id) {
     Pipeline *egress_mau = this->get_pipeline("egress");
     egress_mau->apply(packet.get());
     deparser->deparse(packet.get());
+    
+    if (port == PSA_PORT_RECIRCULATE) {
```

Comment:
Currently this does a packet copy whenever we recirculate a packet. This is very similar to how simple_switch handles recirculation. Seems like its a possibility to remove the packet copying? Thoughts?

Reply:
This comment isn't helpful, other than to say I do not know the answer here.  I am hoping @antoninbas might have time to look at this question and say what he knows of it?

Reply:
By the way you a `reset_metadata` but aren't you supposed to preserve the recirculation metadata?

Reply:
I think the idea is to get recirculate working first, without preserving metadata, and to save for later whatever changes are required to get recirculate while preserving metadata working.

Reply:
Yes that is the plan.